Analysis/Tasks/vertexerhf_opt.cxx

```diff
@@ -120,6 +128,48 @@ struct HFTrackIndexSkimsCreator {
                      pvec0[0], pvec0[1], pvec0[2], track_0.y(),
                      track_1.globalIndex(), pvec1[0], pvec1[1], pvec1[2], track_1.y(),
                      mass_, masssw_);
+	for (auto it2 = it0 + 1; it2 != tracks.end(); ++it2) {
+          if(it2 == it0 || it2 == it1)
+            continue; 
+	  auto& track_2 = *it2;
+          if (abs(track_2.signed1Pt())<3)
+	    continue;
+          UChar_t clustermap_2 = track_2.itsClusterMap();
+          bool isselected_2 = track_2.tpcNClsFound() > 70 && track_2.flags() & 0x4;
+          isselected_2 = isselected_2 && (TESTBIT(clustermap_2, 0) || TESTBIT(clustermap_2, 1));
+          if (!isselected_2)
+            continue;
+          if (track_1.signed1Pt() * track_2.signed1Pt() > 0)
+            continue;
+	  //LOGF(info, abs(track_2.signed1Pt()));
+	  float x2_ = track_2.x();
+          float alpha2_ = track_2.alpha();
+          std::array<float, 5> arraypar2 = {track_2.y(), track_2.z(), track_2.snp(),
+                                            track_2.tgl(), track_2.signed1Pt()};
+          std::array<float, 15> covpar2 = {track_2.cYY(), track_2.cZY(), track_2.cZZ(),
+                                           track_2.cSnpY(), track_2.cSnpZ(),
+                                           track_2.cSnpSnp(), track_2.cTglY(), track_2.cTglZ(),
+                                           track_2.cTglSnp(), track_2.cTglTgl(),
+                                           track_2.c1PtY(), track_2.c1PtZ(), track_2.c1PtSnp(),
+                                           track_2.c1PtTgl(), track_2.c1Pt21Pt2()};
+          o2::track::TrackParCov trackparvar2(x2_, alpha2_, arraypar2, covpar2);
+          df3.setUseAbsDCA(true);
+          int nCand3 = df3.process(trackparvar0, trackparvar1, trackparvar2);
+          if (nCand3 == 0)
+            continue;
+          const auto& vtx3 = df3.getPCACandidate();
+	  //LOGF(info, "test");
+          //LOGF(info, "vertex x %f", vtx3[0]);
+          o2::track::TrackParCov trackdec0_3p = df3.getTrack(0);
+          o2::track::TrackParCov trackdec1_3p = df3.getTrack(1);
+          o2::track::TrackParCov trackdec2_3p = df3.getTrack(2);
```

Comment:
You actually don't need these intermediate assignments, unless you are going to modify the tracks extracted from the df3.